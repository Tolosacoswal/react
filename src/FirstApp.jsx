
import { PropTypes } from "prop-types";

export const FirstApp = ({
    texto,
     subTitle,
     nombre 
    })=>{
    // console.log(props);
   
    return (
        
        <>
        <h1>{ texto}</h1>
    {/* <code>{JSON.stringify(NewMensajeichon)}</code> */}
        <p>{subTitle }</p>
        <p>{nombre }</p>

        </>
    )
    
    
}
FirstApp.prototype={
    texto: PropTypes.string.isRequired, 
    subTitle:PropTypes.string,
}
FirstApp.defaultProps = {
    texto: "no tenemos ningun texto",
    subTitle: "no tenemos texto",
    nombre: "edgard"
}
