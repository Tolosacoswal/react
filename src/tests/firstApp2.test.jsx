import {  getAllByText, render,screen } from "@testing-library/react";

//import { renderIntoDocument } from "react-dom/test-utils";
import { FirstApp } from "../FirstApp";

describe('pruebas en <FirstApp/>', () => { 
  const texto = "hola soy, luffy";
  const subTitle = "el rey de los piratas";
  test('debe de hacer match con el snapshot', () => { 
  const {container}= render (<FirstApp texto/>)
  expect (container).toMatchSnapshot();
});
test('debe de mostrar el mensaje "el rey de los piratas"', () => { 
  render(<FirstApp texto={texto}/>)
  screen.debug();
});
test('debe de mostrar el titulo en un h1', () => {
  render(<FirstApp texto={texto}/>);
  expect( screen.getByRole("heading",{level:1}).innerHTML).toContain(texto);
});
test('debe de mostrar el subtitulo enviado por props', () => { 
  render(<FirstApp 
    texto = {texto}
    subTitle={subTitle}
    />);
    expect(screen.getAllByText(subTitle).length).toBe(2);
  
 });
}); 


   
