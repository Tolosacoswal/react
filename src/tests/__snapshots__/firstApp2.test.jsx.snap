// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <FirstApp/> debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-texto"
  />
  <p>
    no tenemos texto
  </p>
  <p>
    no tenemos texto
  </p>
  <p>
    edgard
  </p>
</div>
`;
